packages/core/src/root.ts

```diff
@@ -360,6 +382,16 @@ export function createRemoteRoot<
           children: value.children.map((child) => serialize(child as any)),
         };
   }
+
+  function validateAllowedComponent(
+    componentType: AllowedComponents,
+    allowedComponents: readonly AllowedComponents[],
+  ) {
+    const isAllowed =
+      allowedComponents.length === 0 ||
```

Comment:
I know this is a little pedantic, but this technically prevents passing an empty array to signal "no components are allowed". If you take my suggestion to always allow text nodes, there might be a use case for differentiating "no `components` option passed" (=== allow all components) and "empty `components` option passed" (=== only text nodes are allowed).

Reply:
Fixed!